slangpy/tests/slangpy_tests/test_tensor.py

```
@@ -277,5 +278,54 @@ def test_primaltensor_sample_scalar(device_type: DeviceType):
+    data = np.array([1.0, 2.0], dtype=np.float32)
+    with pytest.warns(DeprecationWarning, match="deprecated"):
+        tensor = Tensor.numpy(device, data)
+    result = tensor.to_numpy()
+    assert np.allclose(result, data)
+
+
+@pytest.mark.parametrize("device_type", helpers.DEFAULT_DEVICE_TYPES)
+def test_tensor_empty_no_dtype(device_type: DeviceType):
+    """Tensor.empty() with no dtype should raise ValueError."""
+    device = helpers.get_device(device_type)
+    with pytest.raises(ValueError, match="Element type .* must be specified"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)